docker/entrypoint.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# This script creates the necessary files before starting Celestia-appd
+
+# only create the priv_validator_state.json if it doesn't exist and the command is start
+if [[ $1 == "start" && ! -f ${CELESTIA_HOME}/data/priv_validator_state.json ]]
+then
+    mkdir ${CELESTIA_HOME}/data # it is alright if it fails, the script will continue executing
+    cat <<EOF > ${CELESTIA_HOME}/data/priv_validator_state.json
+{
+  "height": "0",
+  "round": 0,
+  "step": 0
+}
+EOF
+fi
+
+/bin/celestia-appd --home ${CELESTIA_HOME} $@
```

Comment:
I'd prefer something like this 
```
if [[ $1 == "celestia-appd" && $2 == "start" && ! -f ${CELESTIA_HOME}/data/priv_validator_state.json ]]. then
    # init
    exec "$@" "--"
elif [ "$1" = 'celestia-appd' ]; then
    exec "$@" "--"
fi
exec "$@"
```
alongside the suggested Dockerfile changes. 

see: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint

This results in a default command of `/bin/celestia-appd`and correctly captures flags. 

See [example usage by ephemeral cluster for ethermint](https://github.com/celestiaorg/ephemeral-cluster/blob/main/docker/ethermint/ethermint-docker-compose.yml#L7)